collector/aws/neptune.go

```diff
@@ -0,0 +1,258 @@
+package aws
+
+import (
+	"finala/collector"
+	"finala/collector/config"
+	"finala/expression"
+	"fmt"
+	"time"
+
+	awsClient "github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cloudwatch"
+	"github.com/aws/aws-sdk-go/service/neptune"
+	"github.com/aws/aws-sdk-go/service/pricing"
+	log "github.com/sirupsen/logrus"
+)
+
+// NeptuneClientDescriptor interface defines the AWS Neptune client
+type NeptuneClientDescriptor interface {
+	DescribeDBInstances(*neptune.DescribeDBInstancesInput) (*neptune.DescribeDBInstancesOutput, error)
+	ListTagsForResource(*neptune.ListTagsForResourceInput) (*neptune.ListTagsForResourceOutput, error)
+}
+
+// NeptuneManager describes the Manager for Neptune
+type NeptuneManager struct {
+	collector          collector.CollectorDescriber
+	client             NeptuneClientDescriptor
+	cloudWatchClient   *CloudwatchManager
+	pricingClient      *PricingManager
+	metrics            []config.MetricConfig
+	region             string
+	namespace          string
+	servicePricingCode string
+	Name               string
+}
+
+// DetectedAWSNeptune defines the detected AWS Neptune instances
+type DetectedAWSNeptune struct {
+	Metric       string
+	Region       string
+	InstanceType string
+	MultiAZ      bool
+	Engine       string
+	collector.PriceDetectedFields
+}
+
+// NewNeptuneManager implements AWS GO SDK
+func NewNeptuneManager(collector collector.CollectorDescriber, client NeptuneClientDescriptor, cloudWatchClient *CloudwatchManager, pricing *PricingManager, metrics []config.MetricConfig, region string) *NeptuneManager {
+	return &NeptuneManager{
+		collector:          collector,
+		client:             client,
+		cloudWatchClient:   cloudWatchClient,
+		pricingClient:      pricing,
+		metrics:            metrics,
+		region:             region,
+		namespace:          "AWS/Neptune",
+		servicePricingCode: "AmazonNeptune",
+		Name:               fmt.Sprintf("%s_neptune", ResourcePrefix),
+	}
+}
+
+// Detect checks which Neptune instance  is under-utilized
+func (np *NeptuneManager) Detect() ([]DetectedAWSNeptune, error) {
+
+	log.WithFields(log.Fields{
+		"region":   np.region,
+		"resource": "neptune",
+	}).Info("starting to analyze resource")
+
+	np.collector.UpdateServiceStatus(collector.EventCollector{
+		ResourceName: np.Name,
+		Data: collector.EventStatusData{
+			Status: collector.EventFetch,
+		},
+	})
+
+	detected := []DetectedAWSNeptune{}
+	instances, err := np.DescribeInstances(nil, nil)
+	if err != nil {
+		log.WithField("error", err).Error("could not describe any neptune instances")
+
+		np.collector.UpdateServiceStatus(collector.EventCollector{
+			ResourceName: np.Name,
+			Data: collector.EventStatusData{
+				Status:       collector.EventError,
+				ErrorMessage: err.Error(),
+			},
+		})
+
+		return detected, err
+	}
+
+	now := time.Now()
+	for _, instance := range instances {
+
+		log.WithField("name", *instance.DBInstanceIdentifier).Info("checking Neptune instances")
```

Comment:
Can you change the log level to `Debug`?

Reply:
Done.